pyccl/baryons/baccoemu_baryons.py

```diff
@@ -0,0 +1,136 @@
+            k (:obj:`float` or `array`): Wavenumber (in :math:`{\\rm Mpc}^{-1}`).
+            a (:obj:`float` or `array`): Scale factor.
+
+        Returns:
+            :obj:`float` or `array`: Correction factor to apply to
+                the power spectrum.
+        """ # noqa
+
+        if hasattr(a, '__len__'):
+            emupars = dict(
+                omega_cold = np.full((len(a)), cosmo['Omega_c'] + cosmo['Omega_b']),
+                omega_baryon = np.full((len(a)), cosmo['Omega_b']),
+                ns = np.full((len(a)), cosmo['n_s']),
+                hubble = np.full((len(a)), cosmo['h']),
+                neutrino_mass = np.full((len(a)), np.sum(cosmo['m_nu'])),
+                w0 = np.full((len(a)), cosmo['w0']),
+                wa = np.full((len(a)), cosmo['wa']),
+                expfactor = a
+            )
+        else:
+            emupars = dict(
+                omega_cold = cosmo['Omega_c'] + cosmo['Omega_b'],
+                omega_baryon = cosmo['Omega_b'],
+                ns = cosmo['n_s'],
+                hubble = cosmo['h'],
+                neutrino_mass = np.sum(cosmo['m_nu']),
+                w0 = cosmo['w0'],
+                wa = cosmo['wa'],
+                expfactor = a
+            )
+
+        if np.isnan(cosmo['A_s']):
+            sigma8tot = cosmo['sigma8']
+            sigma8cold = self._sigma8tot_2_sigma8cold(emupars, sigma8tot)
+            if hasattr(a, '__len__'):
+                emupars['sigma8_cold'] = np.full((len(a)), sigma8cold)
+            else:
+                emupars['sigma8_cold'] = sigma8cold
+        else:
+            if hasattr(a, '__len__'):
+                emupars['A_s'] = np.full((len(a)), cosmo['A_s'])
+            else:
+                emupars['A_s'] = cosmo['A_s']
+
+        _, fka = self.mpk.get_baryonic_boost(k=k / cosmo['h'], **{**emupars, **self.bcm_params})
+        
+        return fka
+
+    def update_parameters(self, log10_M_c=None, log10_eta=None, log10_beta=None, log10_M1_z0_cen=None, 
+                          log10_theta_out=None, log10_theta_inn=None, log10_M_inn=None):
+        """Update BCM parameters. All parameters set to ``None`` will
+        be left untouched.
+        """
+        if log10_M_c is not None:
+            self.bcm_params.update({'M_c' : log10_M_c})
+        if log10_eta is not None:
+            self.bcm_params.update({'eta' : log10_eta})
+        if log10_beta is not None:
+            self.bcm_params.update({'beta' : log10_beta})
+        if log10_M1_z0_cen is not None:
+            self.bcm_params.update({'M1_z0_cen' : log10_M1_z0_cen})
+        if log10_theta_out is not None:
+            self.bcm_params.update({'theta_out' : log10_theta_out})
+        if log10_theta_inn is not None:
+            self.bcm_params.update({'theta_inn' : log10_theta_inn})
+        if log10_M_inn is not None:
+            self.bcm_params.update({'M_inn' : log10_M_inn})
```

Comment:
if `bcm_params.update` is slow in any way, would it make sense to collect all variables to be updated into a single dictionary and only call it once?

Reply:
Yeah, I made this more general now